### ISSUE TITLE
Hide widget JUST in Editor Panel

### ISSUE THREAD

**Issue by thomasmalley** [elementor/elementor#7239]
- [x ] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x ] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**
I can not hide unwanted Widgets in the Editor Panel only.

**Describe the solution you'd like**
I want to hide unwanted Widgets in the Editor Panel only. If necessary via custom PHP code modifying Elementor Plugin directly or a self developed plugin. 

**Describe alternatives you've considered**
I've used the code Posted by @stratboy in #3136 but it doesn't just remove the widgets from the Editor Panel, it removes the widget completely so that any content created using this widget will also be gone. 

**Additional context**
I used the following code for a Plugin:
```php
<?php
/*
Plugin Name: XXX-hide-elementor-widgets
Plugin URI: XXX
Description: Plugin to hide chooses Widgets in the Elementor Editor View
Author: XXX
Author URI: XXX
Version: 1.1
*/
if ( ! defined( 'ABSPATH' ) ) exit;

global $elementor_widget_blacklist;

$elementor_widget_blacklist = [

  'heading'
 
];

add_action('elementor/widgets/widgets_registered', function($widgets_manager){
  global $elementor_widget_blacklist;

  foreach($elementor_widget_blacklist as $widget_name){
    $widgets_manager->unregister_widget_type($widget_name);
  }
}, 15);
?>
```

![screenshot1_before](https://user-images.githubusercontent.com/13106168/53484758-d1b67300-3a84-11e9-84b7-5a376fb9dfa5.png)
![screenshot2_after](https://user-images.githubusercontent.com/13106168/53484759-d24f0980-3a84-11e9-9233-0287d55c44aa.png)

Please please please give a detailed Answer and don't close down this Request immediately  since the Possibility to simplify the Backend is essential for Client Work! Yes I know there is the "new role manager" but it doesn't solve my Problem: I want to give the Client Access to a FEW selected Widgets that he/she can use to add and re-arrange basic Content. 

Best Regards

**Comment by tristanleboss** [comment]
I like the idea of being able to hide some widgets for some users.
The only problem I see is what happens to the instances of the widget already on the page? SHould they be locked so they can't be edited/removed?

**Comment by thomasmalley** [comment]
@tristanleboss indeed 🤔 this is something to think about. I'm glad however that I'm not the only Person who would love to see that Functionality 😅

@Developers is this something you have planned for a more in depth "Role Manager"? 


**Comment by bainternet** [comment]
@thomasmalley 	

If you just want to hide the widget form showing in the panel, you can simply set the widget's `show_in_panel` parameter to `false` ex:

```PHP
add_filter( 'elementor/editor/localize_settings', function( $settings ) {
	$settings['widgets']['heading']['show_in_panel'] = false;
	return $settings;
} );
```

as for a UI to manage this please follow #1314 

thanks.

**Comment by MI808** [comment]
@bainternet What file are you adding this code to? And can you make this specific to certain user roles?

Thanks

**Comment by bdkoder** [comment]
@thomasmalley,
### Here is the solution - 
### In this solution, the widget/widgets will be hide only from Editor Panel. Will works fine within the page

```php
global $elementor_widget_blacklist;

$elementor_widget_blacklist = [
    'heading',
    'your-widget'
];

add_filter('elementor/editor/localize_settings', function ($settings) {
    global $elementor_widget_blacklist;

    // uncomment to list all registered widgets, for debuging only
    // foreach ($settings['initial_document']['widgets'] as $key => $setting) {
    // 	var_dump($key);
    // }

    foreach ($elementor_widget_blacklist as $widget_name) {
        $settings['initial_document']['widgets'][$widget_name]['show_in_panel'] = false;
    }

    return $settings;
}, 99);
```

**Comment by nicholaszein** [comment]
Hello!

### 📢 We're happy to announce that the issue you raised was resolved in Elementor v3.18.0, and improved with Elementor Pro 3.19.0! 🥳 

You can now use the Element Manager to decide what widgets should appear in the Editor and your website, and with Elementor Pro you can decide who can use those widgets. To learn more check the [documentation in the Help Center](https://elementor.com/help/element-manager/).

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
